master_controller:
    ros__parameters:
        verbose: true                                              # printing info about controller states if true
        goal_pose_pub_topic: /master/target                        # topic with goal_pose for nav2
        goal_pose_sub_topic: /goal_pose
        nav2_cmd_vel_topic: /master/warper_cmd_vel                 # output topic for drive_controller
        cmd_vel_topic: /master/cmd_vel                             # output topic for drive_controller
        use_emergency_stop: true                                   # robot stops near dynamic objects if true
        backward_linear_vel: -0.20                                 # linear x velocity for moving backward in recovery
        allow_driving_topic: /master/allow_driving
        feedback_topic: /navigate_to_pose/_action/feedback
        result_topic: /master_controller/result
        odom_topic: /master/odom
        aruco_topic: /aruco/found
        delta: 0.20

dispatcher:
    ros__parameters:
        verbose: true                                              # printing info about controller states if true
        goal_pose_topic: /goal_pose                                # topic with goal_pose for controller
        master_allow_driving_topic: /master/allow_driving
        slave_allow_driving_topic: /slave/allow_driving
        result_topic: /master_controller/result
        aruco_topic: /aruco/found
        slave_result_topic: /slave/result
        slave_goal_pose_topic: /slave/target
        
        #         x     y      t   id   parent_id
        points: [+0.0, +0.0, +0.00,  0.,  0.,  # 0
                 +0.5, +1.5, +1.57,  1.,  0.,  # 1
                 +0.5, +2.5, +1.57,  2.,  1.,  # 2
                 +2.5, +2.5, +0.00,  3.,  2.,  # 3
                 +2.5, +0.5, -1.57,  4.,  3.,  # 4  
                 +1.5, +0.5, +3.14,  5.,  4.,  # 5
                 +2.5, +0.5, +0.00,  6.,  3.,  # 6
                 +2.5, +2.5, +1.57,  7.,  2.,  # 7
                 +0.5, +2.5, +3.14,  8.,  1.,  # 8
                 +0.5, +1.5, +4.71,  9.,  0.,  # 9
                 -1.5, +1.5, +3.14,  10., 1.,  # 10
                 -1.5, +2.5, +1.57,  11., 10., # 11
                 -1.5, +0.5, +4.71,  12., 10., # 12
                 -2.5, +0.5, +3.14,  13., 12., # 13
                 ]

        # #         x     y      t    id   parent_id
        # points: [+0.0, +0.0, +0.00, 0.,  0.,  # 0  d
        #          +0.9, +0.0, +0.00, 1.,  0.,  # 1  d
        #          +1.8, +0.0, +0.00, 2.,  1.,  # 2  d
        #          +1.8, -2.0, -1.57, 3.,  2.,  # 3  d
        #          +0.1, -2.0, +3.14, 4.,  3.,  # 4  d
        #          +0.1, -1.0, +3.14, 5.,  4.,  # 5  d
        #          +0.1, -2.0, -1.57, 6.,  3.,  # 6  d
        #          +1.8, -2.0, +0.00, 7.,  2.,  # 7  d
        #          +1.8, +0.0, +1.57, 8.,  1.,  # 8  d
        #          +0.9, +0.0, +1.57, 9.,  0.,  # 9  d
        #          +0.7, +1.8, +1.57, 10., 1.,  # 10 d
        #          -0.1, +1.8, +3.14, 11., 10., # 11 d
        #          -0.1, +2.6, +1.57, 12., 11., # 12 d
        #          -0.1, +1.8, -1.57, 13., 10., # 13 d
        #          +1.8, +1.8, +0.00, 14., 10., # 14 d
        #          +1.8, +1.1, -1.57, 15., 14., # 15 d
        #          +2.9, +1.1, +0.00, 16., 15., # 16 d
        #          +2.6, +3.1, +1.57, 17., 16., # 17 d
        #          +2.9, +1.1, -1.57, 18., 15., # 18 d
        #          +3.8, +1.2, +0.00, 19., 15., # 19 d
        #          +4.1, -0.5, -1.57, 20., 19., # 20 d
        #          +3.2, -0.6, +3.14, 21., 2.,  # 21 d
        #          ]
